js/canvas.ts

```diff
@@ -879,16 +867,30 @@ export class VideoModes2D implements VideoModes {
     an3State: boolean;
     doubleHiresMode: boolean;
 
-    _flag = 0;
-    _monoMode = false;
+    flag = 0;
+    monoMode = false;
+
+    context: CanvasRenderingContext2D;
 
     constructor(
-        private canvas: HTMLCanvasElement,
+        private screen: HTMLCanvasElement,
         private e: boolean
     ) {
-        this._context = this.canvas.getContext('2d');
-        this._left = (this.canvas.width - 560) / 2;
-        this._top = (this.canvas.height - 384) / 2;
+        this._canvas = document.createElement('canvas');
+        const context = this._canvas.getContext('2d');
+        const screenContext = this.screen.getContext('2d');
+        if (!context || !screenContext) {
+            throw new Error('No 2d context');
+        }
+        this.context = context;
+
+        const { width, height } = { width: 560, height: 192 };
+        this._canvas.width = width;
+        this._canvas.height = height;
+
+        this._screenContext = screenContext;
+        this._left = (this.screen.width - 560) / 2;
+        this._top = (this.screen.height - 384) / 2;
```

Comment:
I don't understand what `_left` and `_top` are meant to do here. `_left` is always 0 and `_top` is `-96`.

Reply:
I get 16, 16 - it's the offset for centering the non-WebGL image because I made the screen canvas larger for the WebGL side of things, so it can expand outside the old 560x384 region for various effects.

Reply:
Ahh, I got confused with the `const { width, height }` assignment above.